### PR TITLE
Upgrade Python 3.11

### PR DESCRIPTION
**Pull request by dmichaels-harvard**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -7,6 +7,12 @@ foursight-smaht
 Change Log
 ----------
 
+0.1.0
```
Comment (netsettler) [inline CHANGELOG.rst:10]: Higher in file than I can easily attach a comment, the ==='s are too short over and under foursight-smaht.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,6 +7,12 @@ foursight-smaht
...
+=====
+* Update to Python 3.11.
```
Comment (netsettler) [inline CHANGELOG.rst:13]: It's a good idea to do a blank line above a bullet list. Some tools don't recognize it if it's pushed up against another paragraph.

File: pyproject.toml [hunk 3]
```diff
@@ -10,24 +10,27 @@ packages = [
 
 [tool.poetry.dependencies]
 # Cannot go to Python 3.8 because cgap-pipeilne is stuck on Python 3.7.
-python = ">=3.7.1,<3.8"
-dcicutils = "^7.5.1"
+python = ">=3.7.1,<3.12"
```
Comment (netsettler) [inline pyproject.toml]: Aren't we dropping 3.7?

**Review by netsettler — APPROVED** [review]
main-ci.yml workflow needs to be in this patch to make it match the python version.
also i think we're dropping 3.7
but this is otherwise fine

**Review by willronchetti — APPROVED** [review]

File: CHANGELOG.rst (hunk 1 above)
Comment (dmichaels-harvard) [reply CHANGELOG.rst:10]: Fixed

File: pyproject.toml (hunk 3 above)
Comment (dmichaels-harvard) [reply pyproject.toml]: Yes, done

File: CHANGELOG.rst (hunk 2 above)
Comment (dmichaels-harvard) [reply CHANGELOG.rst:13]: Fixed

**Pull request merged**